File: app/schemas/forecast.py
from datetime import date
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class ForecastMonthResponse(BaseModel):
    month: date
    income: Decimal
    fixed_expenses: Decimal
    obligation_payments: Decimal
    projected_buffer: Decimal
    has_negative_buffer: bool

    model_config = ConfigDict(from_attributes=True)


class ForecastResponse(BaseModel):
    rows: list[ForecastMonthResponse] = Field(default_factory=list)

File: mobile/src/types/forecast.ts
export interface ForecastMonthResponse {
  month: string;
  income: string;
  fixed_expenses: string;
  obligation_payments: string;
  projected_buffer: string;
  has_negative_buffer: boolean;
}

export interface ForecastResponse {
  rows: ForecastMonthResponse[];
}

export interface ForecastParams {
  start_month: string;
  months: number;
}
